CONTRIBUTING.md

```diff
@@ -146,34 +146,89 @@ The process described here has several goals:
 
 Please follow these steps, detailed in the [pull request template](.github/pull_request_template.md), to have your contribution considered by the maintainers:
 
-1. Describe in your PR what you have done, what this is fixing or introducing and why this isn't regressive. 
-If there is an issue to your PR, please link it. 
+1. Fill out the [pull request template](.github/pull_request_template.md).
+Be sure to tick all boxes that apply to your pull request.  
 2. Follow the [style guides](#style-guides).
-3. After you submit your pull request, verify that all [status checks](https://help.github.com/articles/about-status-checks/) are passing.
-Should a status check fail and you believe it is unrelated to your contribution, please write so in a comment in the PR. 
-
-While the prerequisites above may be satisfied, the reviewer(s) may still ask you to do additional work or ask for changes before your PR is finally accepted. 
-Since the reviewer knows the code very well, they might have ideas how to include features more elegantly or see inconsistencies compared to the code base. 
+3. Describe in your PR what you have done, what this is fixing or introducing and why this isn't regressive.
+If there is an issue to your PR, please link it.
+If you have adhered to the style guide of the [commit messages](#git-commit-messages), you will have that content already.
+4. After you submit your pull request, verify that all [status checks](https://help.github.com/articles/about-status-checks/) are passing.
+Should a status check fail and you believe it is unrelated to your contribution, please write so in a comment in the PR.
+
+While the prerequisites above may be satisfied, the reviewer(s) may still ask you to do additional work or ask for changes before your PR is finally accepted.
+Since the reviewer knows the code very well, they might have ideas how to include features more elegantly or see inconsistencies compared to the code base.
+
+#### Reviews
+Once your pull request is being reviewed, there are some guidelines.
+1. Threads started during a review should only be marked as resolved by the starter of the thread.
+2. Comments and discussions should be ended cleanly. This can be done by using emoticons on the comments,
+e. g. a "thumbs-up" to mark a comment as done.
+3. Once you have integrated changes that were requested, you can dismiss the review of the reviewer and re-request a review.
+
+When integrating a reviewers comments into a pull request, please fix the issue in the respective commit where it appears,
+instead of in a new commit. 
+
+An approval of the PR means that the PR can be merged as is, without any changes still necessary.
+If the owner of a PR is an Eclipse committer of the project, the owner can merge their own PR once approved.
+Otherwise, the PR needs to be merged by an Eclipse committer of the project.
+
+#### Squashing
```

Comment:
Do we want to say, that formatting changes should be in independent commits?

Reply:
Hm. I don't think anyone will do formatting changes in a PR like this. 
Maybe I am wrong though...

Reply:
I have seen formatting in many recent PRs (e.g. spaces after `,`). These should in my opinion be kept in a formatting commit